tests/unit/test_nginx_manager.py

```diff
@@ -49,12 +50,13 @@ def test_reset_file_with_existing_files(patch_nginx_manager_path: None):
     assert not tuple(nginx_manager.NGINX_SITES_ENABLED_PATH.iterdir())
 
 
-def test_update_config_with_valid_config(patch_nginx_manager_path: None):
+def test_update_config_with_valid_config(monkeypatch, patch_nginx_manager_path: None):
     """
     arrange: Valid configuration data.
     act: Create configuration files from the data.
     assert: The files are created and has the configurations.
     """
+    monkeypatch.setattr("nginx_manager._load_config", MagicMock())
```

Comment:
`_load_config` has 2 cases, one where it reloads, and one where it restarts
Shouldn't we have a unittest that cover both cases instead of mocking the whole method ?

Reply:
I will fix this in a follow-up PR.